hugolib/node.go

```diff
@@ -144,10 +180,147 @@ type URLPath struct {
 	Section   string
 }
 
+func (n *Node) URL() string {
+	return n.addLangPathPrefix(n.URLPath.URL)
+}
+
+func (n *Node) Permalink() string {
+	return permalink(n.URL())
+}
```

Comment:
I know we do this already on the `Page` struct (see `.Page.Section()`), but I don't like these two funcs (`URL()` & `Permalink()`).  We're embedding `URLPath` into the `Node`, which embeds the `URL` and `Permalink` fields.  Overshadowing those fields with funcs smells bad to me.

We don't need to fix all of Hugo's problems in this PR, but I'd like to see us change the `Node` struct to have `urlPath URLPath` and then access `n.urlPath.URL`.  We'd need to add the `Slug()` and `Section()` funcs to `Node` and update code all over the place, but at least we'd avoid this overloading of embedded fields business and some confusion about how those embedded `URLPath` fields are supposed to be used, esp. with i18n thrown into the mix.